mne/io/edf/edf.py

```diff
@@ -66,6 +68,21 @@ def find_edf_events(raw):
     return raw.find_edf_events()
 
 
+def _edf_events_from_annotations(raw, event_id):
+    """Modify events_from_annotaitons for EDF specifics.
+
+    Modify events_from_annotaitons so that events[:,1] corresponds to
+    the duration of the events instead of the id of the previous event.
+    """
+    events, event_id_ = events_from_annotations(raw, event_id=event_id,
+                                                use_rounding=False)
+    durations = raw.annotations.duration
+    durations = np.array(durations * raw.info['sfreq'], int)
+    durations[durations != 0] -= 1  # XXX
```

Comment:
Ok, this is the tricky one. See my question https://github.com/mne-tools/mne-python/pull/5574#issuecomment-429533658. To me this line should not be here. I'm only doing it for backward compatibility (when using `_sythetize_stim_ch`). But if it were me, I would enter this PR with that and right afterwards open bug repord, remove this line and fix the tests.

'cos using `_synthetize_stim_ch(events)` should work the same for every file type. Unless EDF stores an extra sample, which I don't think is the case.

Reply:
this is weird indeed. I would fix the raw.annotations so if a duration of one sample is given raw.annotations duration is set to 0.

Reply:
This was added by @mbillingr in #1234 at 8e97b65. In this case, we are building the stim channel taking into account onset and duration, which has not been the case for Brainvision (#5460), eeglab (#5591 #4936), fif (#4936). In all those cases, the events were treated as an impulse.

cc: @jona-sassenhagen are you using EDF files? can you take a look at this discussion? 
cc: @teonbrooks

Reply:
@massich just add a comment as it was done before. @mbillingr made this to avoid overlap of following events when generating the stim channel. It's more an artifact from the way we needed to synthesize the stim channel.